src/app/features/todos/todos.store.ts

```diff
@@ -1,30 +1,50 @@
 import { Injectable } from '@angular/core'
 import { ComponentStore } from '@ngrx/component-store'
-import { map, mergeMap, switchMap, tap } from 'rxjs/operators'
+import { debounceTime, mergeMap, pluck, switchMap, tap } from 'rxjs/operators'
 import { Todo } from './models/todo'
 import { TodosService } from './todos.service'
+import { FormGroup } from '@angular/forms'
+import { UiFormField } from '../../ui/form/ui-form-field'
 
 interface TodosState {
   loading: boolean
   saving: boolean
   todos?: Todo[]
+  filter?: string | undefined
```

Comment:
suggestion: `filter?` implicitly has `string | undefined` already? Is this needed?

Reply:
Fixed